types/presto-client/index.d.ts

```diff
@@ -186,12 +186,44 @@ export interface RuntimeStats {
     completedSplits: number;
 }
 
-export interface PrestoError extends Error {
+// https://github.com/prestodb/presto/blob/bf9df7ef991a77e529d268f5d20ae7a8dc6aebc6/presto-client/src/main/java/com/facebook/presto/client/ErrorLocation.java
+export interface ErrorLocation {
+    columnNumber: number;
+    lineNumber: number;
+}
+
+// https://github.com/prestodb/presto/blob/bf9df7ef991a77e529d268f5d20ae7a8dc6aebc6/presto-client/src/main/java/com/facebook/presto/client/FailureInfo.java
+export interface FailureInfo {
+    cause?: FailureInfo;
+    errorLocation?: ErrorLocation;
+    message?: string;
+    stack: string[];
+    suppressed: FailureInfo[];
+    type: string;
+}
+
+// https://github.com/prestodb/presto/blob/bf9df7ef991a77e529d268f5d20ae7a8dc6aebc6/presto-client/src/main/java/com/facebook/presto/client/QueryError.java
+export interface PrestoQueryError {
+    boolean?: boolean;
+    errorCode?: number;
+    errorLocation?: ErrorLocation;
+    errorName?: string;
+    errorType?: string;
+    failureInfo?: FailureInfo;
+    message: string;
+    sqlState?: string;
+}
+
+// https://github.com/tagomoris/presto-client-node/blob/42a7ca05220a8b6476c68dbecb1a510ed1be5139/lib/presto-client/index.js#LL235C17-L235C31
```

Comment:
There's also the following two ways for the error callback can be called with a custom object that differs from this definition:
* https://github.com/tagomoris/presto-client-node/blob/42a7ca05220a8b6476c68dbecb1a510ed1be5139/lib/presto-client/index.js#L268
* https://github.com/tagomoris/presto-client-node/blob/42a7ca05220a8b6476c68dbecb1a510ed1be5139/lib/presto-client/index.js#L327

I'd make `code` and `error` optional along with `data?: any`. Very messy!